[JIT] Resolve custom classes in source importer

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32938 [WIP][quantization] Use torchbind for Linear PackedParams
* **#32977 [JIT] Resolve custom classes in source importer**
* #32948 [c10] Allow taking a std::tuple as arg

Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

Commit:
[JIT] Resolve custom classes in source importer

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit c6f9ddae:


* **6/6** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=c6f9ddae8c5148767d908c1b71a91948acfeb00e).

### :detective: 5 new failures recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4411077) pytorch_linux_xenial_py3_clang5_mobile_build (1/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95112056) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95112056))
<details>
<summary>
<code>Feb 04 17:52:25 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:64: error: 'getCustomClass' was not declared in this scope</code>
</summary>

```
Feb 04 17:51:55 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/autodiff.cpp.o 
Feb 04 17:51:55 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/attributes.cpp.o 
Feb 04 17:52:00 [ 80%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/argument_spec.cpp.o 
Feb 04 17:52:01 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/custom_class.cpp.o 
Feb 04 17:52:06 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/pass_manager.cpp.o 
Feb 04 17:52:10 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/pickler.cpp.o 
Feb 04 17:52:12 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/unpickler.cpp.o 
Feb 04 17:52:14 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/graph_executor.cpp.o 
Feb 04 17:52:18 [ 81%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o 
Feb 04 17:52:25 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp: In member function 'c10::TypePtr torch::jit::script::SourceImporterImpl::findNamedType(const c10::QualifiedName&)': 
Feb 04 17:52:25 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:64: error: 'getCustomClass' was not declared in this scope 
Feb 04 17:52:25      if (auto custom_class = getCustomClass(name.qualifiedName())) { 
Feb 04 17:52:25                                                                 ^ 
Feb 04 17:52:25 caffe2/CMakeFiles/torch_cpu.dir/build.make:7273: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o' failed 
Feb 04 17:52:25 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o] Error 1 
Feb 04 17:52:25 make[2]: *** Waiting for unfinished jobs.... 
Feb 04 17:52:28 CMakeFiles/Makefile2:903: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Feb 04 17:52:28 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Feb 04 17:52:28 make: *** [all] Error 2 
Feb 04 17:52:28 Makefile:138: recipe for target 'all' failed 
Feb 04 17:52:28 + sccache_epilogue 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4411095) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (2/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95112101) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95112101))
<details>
<summary>
<code>Feb 04 17:52:58 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:29: error: use of undeclared identifier 'getCustomClass'</code>
</summary>

```
Feb 04 17:52:55 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/constant_propagation.cpp.o 
Feb 04 17:52:55 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/constant_pooling.cpp.o 
Feb 04 17:52:55 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/common_subexpression_elimination.cpp.o 
Feb 04 17:52:56 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/create_autodiff_subgraphs.cpp.o 
Feb 04 17:52:56 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_autodiff_subgraphs.cpp.o 
Feb 04 17:52:56 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/insert_guards.cpp.o 
Feb 04 17:52:57 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inliner.cpp.o 
Feb 04 17:52:57 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/lift_closures.cpp.o 
Feb 04 17:52:57 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_forked_closures.cpp.o 
Feb 04 17:52:57 [ 79%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/dead_code_elimination.cpp.o 
Feb 04 17:52:58 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:29: error: use of undeclared identifier 'getCustomClass' 
Feb 04 17:52:58     if (auto custom_class = getCustomClass(name.qualifiedName())) { 
Feb 04 17:52:58                             ^ 
Feb 04 17:52:58 1 error generated. 
Feb 04 17:52:58 caffe2/CMakeFiles/torch_cpu.dir/build.make:18955: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o' failed 
Feb 04 17:52:58 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o] Error 1 
Feb 04 17:52:58 make[2]: *** Waiting for unfinished jobs.... 
Feb 04 17:52:58 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Feb 04 17:52:58 CMakeFiles/Makefile2:5410: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Feb 04 17:52:58 Makefile:138: recipe for target 'all' failed 
Feb 04 17:52:58 make: *** [all] Error 2 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4411093) caffe2_onnx_py2_gcc5_ubuntu16_04_build (3/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95112117) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95112117))
<details>
<summary>
<code>Feb 04 17:53:04 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:64: error: 'getCustomClass' was not declared in this scope</code>
</summary>

```
Feb 04 17:53:02 [ 77%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inliner.cpp.o 
Feb 04 17:53:02 [ 77%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/lift_closures.cpp.o 
Feb 04 17:53:02 [ 77%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_forked_closures.cpp.o 
Feb 04 17:53:03 [ 77%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/dead_code_elimination.cpp.o 
Feb 04 17:53:03 [ 77%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/decompose_ops.cpp.o 
Feb 04 17:53:03 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/canonicalize_ops.cpp.o 
Feb 04 17:53:04 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/erase_number_types.cpp.o 
Feb 04 17:53:04 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/fixup_trace_scope_blocks.cpp.o 
Feb 04 17:53:04 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_fork_wait.cpp.o 
Feb 04 17:53:04 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp: In member function 'c10::TypePtr torch::jit::script::SourceImporterImpl::findNamedType(const c10::QualifiedName&)': 
Feb 04 17:53:04 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:64: error: 'getCustomClass' was not declared in this scope 
Feb 04 17:53:04      if (auto custom_class = getCustomClass(name.qualifiedName())) { 
Feb 04 17:53:04                                                                 ^ 
Feb 04 17:53:04 caffe2/CMakeFiles/torch_cpu.dir/build.make:18955: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o' failed 
Feb 04 17:53:04 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o] Error 1 
Feb 04 17:53:04 make[2]: *** Waiting for unfinished jobs.... 
Feb 04 17:53:05 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Feb 04 17:53:05 CMakeFiles/Makefile2:5410: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Feb 04 17:53:05 make: *** [all] Error 2 
Feb 04 17:53:05 Makefile:138: recipe for target 'all' failed 
Feb 04 17:53:05 Building wheel torch-1.5.0a0+c6f9dda 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4411091) pytorch_linux_xenial_py3_clang5_asan_build (4/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95112152) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95112152))
<details>
<summary>
<code>Feb 04 17:53:42 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:29: error: use of undeclared identifier 'getCustomClass'</code>
</summary>

```
Feb 04 17:53:39 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/insert_guards.cpp.o 
Feb 04 17:53:40 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inliner.cpp.o 
Feb 04 17:53:40 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/lift_closures.cpp.o 
Feb 04 17:53:40 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_forked_closures.cpp.o 
Feb 04 17:53:40 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/dead_code_elimination.cpp.o 
Feb 04 17:53:41 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/decompose_ops.cpp.o 
Feb 04 17:53:41 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/canonicalize_ops.cpp.o 
Feb 04 17:53:41 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/erase_number_types.cpp.o 
Feb 04 17:53:41 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/fixup_trace_scope_blocks.cpp.o 
Feb 04 17:53:42 [ 78%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/inline_fork_wait.cpp.o 
Feb 04 17:53:42 /var/lib/jenkins/workspace/torch/csrc/jit/import_source.cpp:112:29: error: use of undeclared identifier 'getCustomClass' 
Feb 04 17:53:42     if (auto custom_class = getCustomClass(name.qualifiedName())) { 
Feb 04 17:53:42                             ^ 
Feb 04 17:53:42 1 error generated. 
Feb 04 17:53:42 caffe2/CMakeFiles/torch_cpu.dir/build.make:19675: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o' failed 
Feb 04 17:53:42 make[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o] Error 1 
Feb 04 17:53:42 make[2]: *** Waiting for unfinished jobs.... 
Feb 04 17:53:42 CMakeFiles/Makefile2:5104: recipe for target 'caffe2/CMakeFiles/torch_cpu.dir/all' failed 
Feb 04 17:53:42 make[1]: *** [caffe2/CMakeFiles/torch_cpu.dir/all] Error 2 
Feb 04 17:53:42 make: *** [all] Error 2 
Feb 04 17:53:42 Makefile:138: recipe for target 'all' failed 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4411075) pytorch_macos_10_13_py3_build (5/5)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=95112170) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=95112170))
<details>
<summary>
<code>Feb 04 09:54:00 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o</code>
</summary>

```
Feb 04 09:53:58 [2470/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/alias_analysis.cpp.o 
Feb 04 09:53:58 [2471/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/batch_mm.cpp.o 
Feb 04 09:53:59 [2472/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/bailout_graph.cpp.o 
Feb 04 09:53:59 ../torch/csrc/jit/passes/bailout_graph.cpp:122:10: warning: unused variable 'carried_deps' [-Wunused-variable] 
Feb 04 09:53:59     auto carried_deps = lv.carriedInputsWithCond(); 
Feb 04 09:53:59          ^ 
Feb 04 09:53:59 1 warning generated. 
Feb 04 09:53:59 [2473/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/canonicalize.cpp.o 
Feb 04 09:53:59 [2474/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/clear_undefinedness.cpp.o 
Feb 04 09:54:00 [2475/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o 
Feb 04 09:54:00 FAILED: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o  
 -DUSE_AVX2 -DTH_HAVE_THREAD -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wno-write-strings -Wno-unknown-pragmas -Wno-missing-braces -fvisibility=hidden -O2 -DCAFFE2_BUILD_MAIN_LIB -DASMJIT_STATIC -std=gnu++14 -MD -MT caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o -MF caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o.d -o caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/import_source.cpp.o -c ../torch/csrc/jit/import_source.cpp 
Feb 04 09:54:00 ../torch/csrc/jit/import_source.cpp:112:29: error: use of undeclared identifier 'getCustomClass' 
Feb 04 09:54:00     if (auto custom_class = getCustomClass(name.qualifiedName())) { 
Feb 04 09:54:00                             ^ 
Feb 04 09:54:00 1 error generated. 
Feb 04 09:54:00 [2476/3006] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/passes/constant_propagation.cpp.o 
Feb 04 09:54:00 ninja: build stopped: subcommand failed. 
Feb 04 09:54:00 Traceback (most recent call last): 
Feb 04 09:54:00   File "setup.py", line 737, in <module> 
Feb 04 09:54:00     build_deps() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]

Commit:
Update on "[JIT] Resolve custom classes in source importer"


Differential Revision: [D19724588](https://our.internmc.facebook.com/intern/diff/D19724588)

[ghstack-poisoned]
